index.js

```diff
@@ -11,59 +16,47 @@ const tagsDir = path.join(__dirname, './tags');
 const localesDir = path.join(__dirname, './locales');
 const inputServicesDir = path.join(__dirname, './services');
 const outputServicesFile = path.join(assetsDir, 'services.json');
+const servicesI18nFile = path.join(assetsDir, 'services_i18n.json');
 
 /**
- * Validate services JSON file.
+ * Build services data by reading and processing content from a destination JSON file
+ * and source YAML files. Differences are handled, and the final grouped data is written back
+ * to the destination JSON file.
  *
- * @param {string} filePath - The file path for services JSON file.
- * @throws {Error} - If JSON  is not valid.
- */
-const validateJson = async (filePath) => {
-    try {
-        JSON.parse(await fs.readFile(filePath, 'utf8'));
-    } catch (error) {
-        logger.error(`Failed to parse ${filePath}`, error.message);
-        process.exit(1);
-    }
-};
-
-/**
- * Gets the names of YML file from the services folder.
- *
- * @param {string} inputDirPath - The path to the folder with service files.
- * @returns {Promise<Array<string>>} - An array of services file names.
- */
-const getServicesFileNames = async (inputDirPath) => {
-    // get all dir names from services folder
-    const fileNames = await fs.readdir(inputDirPath);
-    // get the file names without its extension
-    const fileBaseNames = fileNames.map((file) => path.parse(file).name);
-    // return sorted array
-    return fileBaseNames.sort();
-};
-
-/**
- * Builds the result services file and saves it to `resultFilePath`.
- * During the build the following steps are performed:
- * 1. Check if the services JSON file is valid.
- * 2. Check if the services in the folder have been deleted by comparing with the data in JSON file.
- * 3. If the information has been deleted, write the missing files.
- * 4. Collect information from the services files, sort and overwrite blocked services files.
+ * @param {string} distFilePath - The file path to the destination JSON file.
+ * @param {string} sourceDirPath - The directory path containing source YAML files.
+ * @param {string} localesFolder - The directory with translations
+ * @param {string} combinedTranslationsFile - The file with combined translation for service groups
+ * @returns {Promise<void>} - A Promise resolving once the build process is complete.
  *
- * @param {string} inputDirPath - The directory path where the services data is located.
- * @param {string} resultFilePath - The file path for the "services.json" file.
- * @returns {Promise<void>} A promise that resolves when the building process is complete.
+ * @throws {Error} - Throws an error if there's an issue during the build process.
  */
-const buildServices = async (inputDirPath, resultFilePath) => {
+const buildServices = async (distFilePath, sourceDirPath, localesFolder, combinedTranslationsFile) => {
```

Comment:
do not change the current interface of the function, no need of passing `localesFolder` and `combinedTranslationsFile` as they are needed only for `addServiceLocalizations` which can be run separately:

```
await buildServices(outputServicesFile, inputServicesDir);
await addServiceLocalizations(localesDir, servicesI18nFile);
```